# Spotify Top 50 Musics in 2025 - Volume vs Consistência

### 1. Carregando o dataset

In [1]:
import pandas as pd
#index_col=0 impede que o pandas crie uma coluna extra de índice
df = pd.read_csv('2025_top50_songs.csv', index_col=0)
df.head()

,song_title,artist,streams_2025_billions,primary_genre,bpm,duration_seconds,release_year,artist_country,explicit,danceability,energy,valence,acousticness,peak_global_chart_position,dataset_part
wrapped_2025_rank,,,,,,,,,,,,,,,
1,Die With A Smile,Lady Gaga & Bruno Mars,1.70,Pop,120,251,2024,USA,False,0.72,0.68,0.55,0.18,1,Spotify Wrapped 2025 Top 50 Global Songs
2,APT.,ROSE & Bruno Mars,1.62,K-Pop/Pop,145,173,2024,South Korea,False,0.86,0.79,0.88,0.05,1,Spotify Wrapped 2025 Top 50 Global Songs
3,Espresso,Sabrina Carpenter,1.55,Pop,104,175,2024,USA,False,0.80,0.73,0.96,0.06,1,Spotify Wrapped 2025 Top 50 Global Songs
4,Please Please Please,Sabrina Carpenter,1.48,Pop,107,186,2024,USA,False,0.70,0.66,0.87,0.10,1,Spotify Wrapped 2025 Top 50 Global Songs
5,Taste,Sabrina Carpenter,1.41,Pop,117,177,2024,USA,True,0.78,0.70,0.82,0.08,1,Spotify Wrapped 2025 Top 50 Global Songs


### 2. Streams totais por artista

In [2]:
#coloquei dentro de uma variavel para fazer a métrica musica/streams
artist_streams = df.groupby('artist')['streams_2025_billions'].sum().sort_values(ascending=False)
artist_streams

,streams_2025_billions
artist,
Sabrina Carpenter,5.24
Billie Eilish,3.26
Chappell Roan,2.72
Taylor Swift,2.32
Shaboozey,2.12
Kendrick Lamar,2.04
Lady Gaga & Bruno Mars,1.70
ROSE & Bruno Mars,1.62
Tyla,1.55


### 3. Músicas por artista

In [3]:
#soma da quantidade de músicas por artista
artist_musics = df['artist'].value_counts()
artist_musics

,count
artist,
Sabrina Carpenter,4
Chappell Roan,3
Billie Eilish,3
Taylor Swift,3
Beyoncé,2
Olivia Rodrigo,2
Kendrick Lamar,2
Shaboozey,2
Gracie Abrams,2


### 4. Métrica de eficiência
Streams totais dividido pelo número de músicas - quem faz mais com menos.

In [4]:
#métrica música/streams, utilizei ascending=False para mostrar do maior para o menor
metrics = artist_streams / artist_musics
metrics.sort_values(ascending=False)

,0
artist,
Lady Gaga & Bruno Mars,1.700000
ROSE & Bruno Mars,1.620000
Sabrina Carpenter,1.310000
Benson Boone,1.240000
Teddy Swims,1.200000
Hozier,1.180000
ROSE,1.120000
Tommy Richman,1.090000
Billie Eilish,1.086667


## 5. Conclusões

* **Sabrina Carpenter** domina em volume absoluto com **5.24 bilhões** de streams,
  resultado direto da consistência com **4 músicas** no ranking. Nenhum outro
  artista chegou perto em streams totais.

* Porém, quando olhamos eficiência (streams por música) a história muda.
  **Lady Gaga & Bruno Mars** lideram com **1.70 bilhões por música**, contra
  **1.31** de Sabrina. Um único hit superou a média de qualquer artista consistente.

* A conclusão é que **as duas estratégias funcionam**, mas de formas diferentes:
  consistência garante volume total, mas um hit certeiro pode ser mais eficiente
  do que várias músicas medianas.